Assets/Scripts/PlayerController.cs

```diff
@@ -34,13 +35,19 @@ public class PlayerController : MonoBehaviour {
     // Main game loop
     private void Update() {
         pointerInput = GetPointerInput();
-        agent.agentMover.movementInput = movementInput;
+        if (rollFrameCounter == 0) { agent.movementDirection = movementInput; } // Don't update if agent is in roll frames
```

Comment:
Instead of the agent's agentMover, update the direction since its getting updated in agent every frame anyways

From me looking at code, looks like ownership should be player controller -> agent -> agentMover. Let me know if wrong too, but it was causing some bugs